### PR TITLE
Correctly escape escape sequences in template literals.

### LINKED ISSUE #79: Escape \u sequences properly (angular2/src/change_detection/parser/lexer.dart)

### PR DESCRIPTION
**Pull request by mprobst**
Fixes #79.

### CODE REVIEW

File: test/DartTest.ts [hunk 1]
```diff
@@ -275,6 +277,10 @@ describe('transpile to dart', () => {
       expectTranslate("`\\$a`").to.equal(" '''\\$a''' ;");
     });
 
+    it('escapes escape sequences', () => {
+      expectTranslate("`\\\\u1234`").to.equal(" '''\\\\u1234''' ;");
```
Comment (alexeagle) [inline test/DartTest.ts]: but we still have a problem in the angular code, right? https://github.com/angular/angular/blob/master/modules/angular2/src/change_detection/parser/lexer.js#L375 doesn't have four backslashes in the file, it has two. Unless the problem is that we have to escape the input here in a way we don't when reading from a file?

Comment (mprobst) [inline test/DartTest.ts]: No, we're all good. This has four backslashes, because it's TypeScript source that is escaped within a TypeScript string, doubling the number of backslashes. The actual string after parsing has two backslashes, like the Angular source. And that then generates a two-backslash result, as expected and tested for.

Yeah, this is all a bit confusing.

Comment (alexeagle) [inline test/DartTest.ts]: Yes, I believe you now. The extra double quotes around the string are the clue.
An e2e test would be easier to read in this respect, though I think this edge case doesn't need such a heavyweight verification.
